crates/common/fork_choice/src/handlers.rs

```diff
@@ -203,3 +205,33 @@ pub async fn on_block(
 
     Ok(())
 }
+
+/// Run ``on_attester_slashing`` immediately upon receiving a new ``AttesterSlashing``
+/// from either within a block or directly on the wire.
+pub fn on_attester_slashing(
+    mut store: Store,
+    attester_slashing: AttesterSlashing,
+) -> anyhow::Result<()> {
+    let attestation_1 = attester_slashing.attestation_1;
+    let attestation_2 = attester_slashing.attestation_2;
+    ensure!(is_slashable_attestation_data(
+        &attestation_1.data,
+        &attestation_2.data
+    ));
+    let state = &store.block_states[&store.justified_checkpoint.root];
+    ensure!(state.is_valid_indexed_attestation(&attestation_1)?);
+    ensure!(state.is_valid_indexed_attestation(&attestation_2)?);
+
+    let mut indices = vec![];
+
+    for i in &attestation_1.attesting_indices {
+        if attestation_2.attesting_indices.contains(i) {
+            indices.push(*i);
+        }
+    }
+
+    for index in indices {
+        store.equivocating_indices.push(index);
+    }
```

Comment:
```suggestion
    let attestation_1_indices = attestation_1
        .attesting_indices
        .into_iter()
        .collect::<HashSet<_>>();
    let attestation_2_indices = attestation_2
        .attesting_indices
        .into_iter()
        .collect::<HashSet<_>>();

    for index in attestation_1_indices.intersection(&attestation_2_indices) {
        store.equivocating_indices.push(*index);
    }
```

Your solution above is the same as

```
for i in att_1 {
    for k in att_2 {
    }
}
```

This solution is O(n^2) which is slow. The solution I suggested is O(n)